lib/app/admin/data-reset/create.js

```diff
@@ -0,0 +1,119 @@
+ * @param {Object} mediator  - mediator object used for communication
+ * @returns {Promise} Promise, resolved if all items are successfully created.
+ */
+// var buildDataDo = function(mediator) {
+//   var deferred = q.defer();
+//   var relationsMap = [];
+//   console.log('>>>>>Start create');
+//   async.eachSeries(DEFAULT_ITEM_LIST, function(itemType, eachTypeCb) {
+//     createItemsForType(mediator, itemType, relationsMap, function createItemsCb(err, mappingsForType) {
+//       relationsMap[itemType] = mappingsForType;
+//       eachTypeCb();
+//     });
+//   }, function(err) {
+//     console.log('>>>>>create end');
+//     (err) ? deferred.reject(err) : deferred.resolve();
+//   });
+//
+//   return deferred.promise;
+// };
+
+
+/**
+ * @callback createItemsCb
+ * @param {Error} err - an error or null
+ * @param {Array} mappingsForType - mapping between seed data-reset id and newly created dada ids
+ */
+
+/**
+ * Creates all items of a given type
+ * @param {Object} mediator  - mediator object used for communication
+ * @param {String} itemType - type of the item to remove e.g. result, workorder
+ * @param {Array} relationsMap - An array mapping out relations between seed and newly created data-reset
+ *
+ * @param {createItemsCb} createItemsCb - callback called after all items for type have been created
+ */
+var createItemsForType = function(mediator, itemType, relationsMap, createItemsCb) {
+  var topic = 'wfm:cloud:' + itemType + ':create';
+  var getSampleData = require('./data-demo/' + itemType);
+
+  var mappingsForType = [];
+  async.eachSeries(getSampleData(), function(dataItem, eachItemCb) {
+    var mapping = {
+      seedId: dataItem.id,
+      newId: ''
+    };
+    dataItem = relationsMapper.mapRelations(itemType, dataItem, relationsMap[DEFAULT_LIST_RELATIONS[itemType]]);
+    createItem(mediator, topic, dataItem, function createSingleItemCb(err, newId) {
+      if (err) {
+        eachItemCb(err);
+      } else {
+        mapping.newId = newId;
+        mappingsForType.push(mapping);
+        eachItemCb();
+      }
+    });
+  }, function(err) {
+    createItemsCb(err, mappingsForType);
+  });
+};
+
+/**
+ * @callback createItemCb
+ * @param {Error} err - error or null
+ * @param {String} newId - Id of created object.
+ */
+
+/**
+ *
+ * @param {Object} mediator  - mediator object used for communication
+ * @param {String} topic - topic prefix to be published
+ * @param {Object} dataItem - single item object to be created e.g workorder, membership
+ * @param {createItemCb} createSingleItemCb - callback that finishes mapping and calls
+ */
+var createItem = function(mediator, topic, dataItem, createSingleItemCb) {
+
+  mediator.request(topic, [dataItem, dataItem.id], {uid: dataItem.id}) //error here
+    .then(function(createdItem) {
+      createSingleItemCb(null, createdItem.id);
+    })
+    .catch(function(err) {
+      console.log('error: ', err, '\ntopic:', topic);
+      createSingleItemCb(new Error(err));
```

Comment:
You already called this in the `then()`. The issue is that if the callback itself throws an error (and you have no control over that), then the error will be propagated to this catch(), and you'll then call the callback a second time, with the error. This is one of the major reasons to avoid mixing promises and callbacks.